Gcc build broken on Ubuntu Focal

It seems that the version of NASM that comes with Ubuntu Focal 20.04 now cares that there is a duplicate output file specified. I would have tried to fix it, but the first `-o iasmsupt.obj` appears to come from the NASMFLAGS variable and I couldn't see where that got set?

~~~
nasm -Dgcc -D_INIT -f obj -DWITHFAT32 -i../hdr/ -DXCPU=86 -felf -o iasmsupt.obj -o iasmsupt.obj asmsupt.asm
iasmsupt.obj: fatal: more than one output file specified: iasmsupt.obj

make[1]: *** [GNUmakefile:161: iasmsupt.obj] Error 1
make[1]: Leaving directory '/clients/common/fdkernel.git/kernel'
~~~

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,12 +1,10 @@
 language: c
 
-dist:  bionic
+dist:  focal
 
 addons:
   apt:
     sources:
-      - sourceline: 'ppa:dosemu2/ppa'
-      - sourceline: 'ppa:stsp-0/djgpp'
       - sourceline: 'ppa:tkchia/build-ia16'
     packages:
       - gcc-ia16-elf

--- a/kernel/makefile
+++ b/kernel/makefile
@@ -158,7 +158,7 @@ initclk.obj: initclk.c  $(INITHEADERS) $(TARGET).lnk
 
 #the string functions for INIT_TEXT
 iasmsupt.obj: asmsupt.asm $(TARGET).lnk
-	$(NASM) -D$(COMPILER) -D_INIT -f obj $(NASMFLAGS) -o iasmsupt.obj asmsupt.asm
+	$(NASM) -D$(COMPILER) -D_INIT -f obj $(NASMFLAGS) -o $@ asmsupt.asm
 
 #the printf for INIT_TEXT - yet another special case, this file includes prf.c
 iprf.obj: iprf.c prf.c $(HDR)portab.h $(TARGET).lnk

--- a/mkfiles/gcc.mak
+++ b/mkfiles/gcc.mak
@@ -25,7 +25,7 @@ NASMFLAGS+=-DWITHFAT32
 endif
 
 NASM=$(XNASM)
-NASMFLAGS+=-i../hdr/ -DXCPU=$(XCPU) -felf -o $@
+NASMFLAGS+=-i../hdr/ -DXCPU=$(XCPU) -felf
 
 CC=ia16-elf-gcc -c
 CL=ia16-elf-gcc
@@ -78,7 +78,7 @@ LINK=$(XLINK) -Tkernel.ld -Wl,-Map,kernel.map -o kernel.exe $(OBJS) -Wl,--whole-
 
 #               *Implicit Rules*
 .asm.obj :
-	$(NASM) -D$(COMPILER) $(NASMFLAGS) $*.asm
+	$(NASM) -D$(COMPILER) $(NASMFLAGS) -o $@ $<
 
 .c.obj :
 	$(CC) $(CFLAGS) $*.c

--- a/travis_build.sh
+++ b/travis_build.sh
@@ -1,5 +1,7 @@
 #!/bin/sh
 
+set -e
+
 if [ -z "${TRAVIS_BUILD_DIR}" ] ; then
   TRAVIS_BUILD_DIR=$(pwd)
 fi

